internal/mode/static/state/graph/route_common.go

```diff
@@ -529,6 +535,26 @@ func routeAllowedByListener(
 	return true
 }
 
+// isRouteKindAllowedByListener checks if the route kind is allowed by the listener.
+// If the listener does not specify allowed kinds, all kinds can attach to it.
+// If the listener specifies allowed kinds, the route kind must be in the list.
+// If the listener specifies HTTPRoute, a GRPCRoute can be attached to it.
```

Comment:
Why is this the case?

Reply:
gRPC uses HTTP/2 as its transport protocol, the underlying network components (such as the Gateway listeners) only need to support HTTP/2 to handle gRPC traffic. This compatibility ensures that HTTPRoute kinds can encompass gRPC routes.

Reply:
A GRPCRoute should not be able to attach to a listener that allows only HTTPRoutes.

Reply:
The description of this [bug](https://github.com/nginxinc/nginx-gateway-fabric/issues/2299) suggests otherwise

```
Note: GRPCRoutes can also be attached to Gateway listeners that only allow kinds HTTPRoute.
```

Reply:
If the user configures the following listener:

```
- name: http
  port: 80
  protocol: HTTP
  allowedRoutes:
    namespaces:
      from: Same
    kinds:
      - kind: "HTTPRoute"
```

then we can only permit HTTPRoutes to attach to the listener. 


If, however, the user does not specify allowedRoutes.kinds:

```
- name: http
  port: 80
  protocol: HTTP
  allowedRoutes:
    namespaces:
      from: Same
```

then we can infer from the protocol which routes to allow. In the case of HTTP or HTTPS, I think it is correct to allow both HTTPRoutes and GRPCRoutes. However, this is only the case when the allowedRoutes.kinds field is not set.

Reply:
@salonichf5 by that note, I meant that the bug also applies to GRPCRoutes.I apologize if that note was misleading.